cv_next/server/api/cvs.ts

```diff
@@ -158,38 +158,77 @@ export async function getPaginatedCvs(
     let query = supabase
       .from(Tables.cvs)
       .select("*")
-      .range(from, to - 1);
+      .order(CvKeys.created_at, { ascending: false })
+      .eq(CvKeys.deleted, !filterOutDeleted);
+    let profileQuery = supabase.from(Tables.profiles).select(ProfileKeys.id);
 
     logger.debug(filters, "filters");
-    if (filters) {
-      if (filters.searchValue) {
-        query = query.textSearch(CvKeys.description, filters.searchValue);
-      }
-      if (filters.categoryIds) {
-        logger.debug(filters.categoryIds, "category ids");
-        query = query.overlaps(CvKeys.cv_categories, filters.categoryIds);
-      }
-    }
 
-    if (filterOutDeleted) {
-      query = query.eq(CvKeys.deleted, false);
-    }
+    profileQuery = applyProfileSearchFilter(profileQuery, filters);
+    query = applyCategoryFilter(query, filters);
+    query = await applyProfileSearchToCvs(query, profileQuery, filters);
 
     const { data: cvs, error } = await query;
-    logger.debug(cvs?.map((cv) => cv.cv_categories, "cvs"));
+    logger.debug(
+      cvs?.map((cv) => cv.cv_categories),
+      "cvs"
+    );
 
     if (error) {
       logger.error(error, "getPaginatedCvs");
       return null;
     }
 
-    return { page: page, cvs: cvs as CvModel[] };
+    const uniqueCvs = [...new Map(cvs.map((cv) => [cv.id, cv])).values()].slice(
+      from,
+      to
+    );
```

Comment:
Isn't it expensive to always get everything from the DB and just return a specific amount? Or does that even out(or even outweigh) the price of calling all the filters on just a small amount?

Reply:
Wait wait I didn't notice this code.
Why did we deleted the `range` in the supabase query?
The paginated selection should be in the db, not manually after getting all the results..

Reply:
That's what I thought, I'll move it back up